installer/cli/microk8s.py

```diff
@@ -104,55 +104,62 @@ def show_help():
-        )
-    )
+      --cpu      Cores used by MicroK8s (default={definitions.DEFAULT_CORES}, min={definitions.MIN_CORES})
+      --mem      RAM in GB used by MicroK8s (default={definitions.DEFAULT_MEMORY}, min={definitions.MIN_MEMORY})
+      --disk     Max volume in GB of the dynamically expandable hard disk to be used (default={definitions.DEFAULT_DISK}, min={definitions.MIN_DISK})
+      --channel  Kubernetes version to install (default={definitions.DEFAULT_CHANNEL})
+      -y, --assume-yes  Automatic yes to prompts"""  # noqa
+    Echo.info(msg)
+
+
+def memory(mem_gb: str) -> int:
+    mem_gb = int(mem_gb)
+    if mem_gb < definitions.MIN_MEMORY:
+        raise ValueError("Out of valid memory range")
+    return mem_gb
+
+
+def cpu(cpus: str) -> int:
+    cpus = int(cpus)
+    if cpus < definitions.MIN_CORES:
+        raise ValueError("Invalid number of cpus")
+    return cpus
+
+
+def disk(disk_gb: str) -> int:
+    disk_gb = int(disk_gb)
+    if disk_gb < definitions.MIN_DISK:
+        raise ValueError("Out of valid disk range")
+    return disk_gb
 
 
 def install(args) -> None:
     if "--help" in args or "-h" in args:
         _show_install_help()
         return
+
     parser = argparse.ArgumentParser("microk8s install")
-    parser.add_argument("--cpu", default=definitions.DEFAULT_CORES, type=int)
-    parser.add_argument("--mem", default=definitions.DEFAULT_MEMORY, type=int)
-    parser.add_argument("--disk", default=definitions.DEFAULT_DISK, type=int)
+    parser.add_argument("--cpu", default=definitions.DEFAULT_CORES, type=cpu)
+    parser.add_argument("--mem", default=definitions.DEFAULT_MEMORY, type=memory)
+    parser.add_argument("--disk", default=definitions.DEFAULT_DISK, type=disk)
     parser.add_argument("--channel", default=definitions.DEFAULT_CHANNEL, type=str)
     parser.add_argument(
         "-y", "--assume-yes", action="store_true", default=definitions.DEFAULT_ASSUME
     )
-    args = parser.parse_args(args)
-
     echo = Echo()
+    args = parser.parse_args(args)
 
-    if platform == "win32":
-        aux = Windows(args)
-        if not aux.is_enough_space():
-            echo.warning("VM disk size requested exceeds free space on host.")
-
-    if platform == "darwin":
-        aux = MacOS(args)
-        if not aux.is_enough_space():
-            echo.warning("VM disk size requested exceeds free space on host.")
-
-    else:
-        aux = Linux(args)
-        if not aux.is_enough_space():
-            echo.warning("VM disk size requested exceeds free space on host.")
+    host = {"win32": Windows, "darwin": MacOS, "linux": Linux}[platform](args)
+    if not host.has_enough_disk_space():
+        echo.warning("VM disk size requested exceeds free space on host.")
+    if not host.has_enough_cpus():
+        echo.warning("VM cpus requested exceed number of available cores on host.")
+    if not host.has_enough_memory():
+        echo.warning("VM memory requested exceeds the total memory on host.")
```

Comment:
Is it OK to just emit a warning in these cases?

Reply:
I think we need to not proceed with the wrong CPU size. Here is what I got:
```
jackal@aurora:~/workspace/microk8s/installer$ ./microk8s install --cpu 1000
VM cpus requested exceed number of available cores on host.
launch failed: The following errors occurred:                                   
qemu-system-x86_64: Invalid SMP CPUs 1000. The max CPUs supported by machine 'pc-i440fx-focal' is 255

launch failed: instance "microk8s-vm" already exists                            


launch failed: instance "microk8s-vm" already exists                            


launch failed: instance "microk8s-vm" already exists                            


An error occurred with the instance when trying to launch with 'multipass': returned exit code 2.
Ensure that 'multipass' is setup correctly and try again.
``` 

Resulting to a failed installation.

Reply:
Same with more VM memory:
```
jackal@aurora:~/workspace/microk8s/installer$ ./microk8s install --mem 64 
VM memory requested exceeds the total memory on host.
Launched: microk8s-vm                                                           
An error occurred when trying to execute 'sudo ping -c 1 snapcraft.io' with 'multipass': returned exit code 1.
```